src/modules/launcher/Wox.Plugin/Common/VirtualDesktopHelper.cs

```diff
@@ -0,0 +1,411 @@
+
+        /// <summary>
+        /// Instance of "Virtual Desktop Manager"
+        /// </summary>
+        private IVirtualDesktopManager virtualDesktopManager;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VirtualDesktopHelper"/> class.
+        /// </summary>
+        /// <param name="desktopListUpdate">Setting to configure if the list of available desktops should update automatically or only when calling <see cref="UpdateDesktopList"/>. Per default this is set to manual update (false) to have less registry queries.</param>
+        public VirtualDesktopHelper(bool desktopListUpdate = false)
+        {
+            // ToDo: Error-Handling if VDs are diabled and instance can't be creted.
+            // ToDo: Set private var, Log com exception, handle init errors in other pblic methods
+            // ToDo: Exc on init: "Init vdh: Failed to create vdm instance."
+            // ToDo: Exc in other methods: "XYZ: Operation failed with HResult {}. This can happen if the vdm instance isn't available.
+            virtualDesktopManager = (IVirtualDesktopManager)new CVirtualDesktopManager();
+
+            _desktopListAutoUpdate = desktopListUpdate;
+            UpdateDesktopList();
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the <see cref="VirtualDesktopHelper"/> class.
+        /// </summary>
+        ~VirtualDesktopHelper()
+        {
+            virtualDesktopManager = null;
+            availableDesktops = null;
+            currentDesktop = Guid.Empty;
+        }
+
+        /// <summary>
+        /// Method to update the list of Virtual Desktops from Registry
+        /// The data in the registry are always up to date
+        /// </summary>
+        public void UpdateDesktopList()
+        {
+            // List of all desktops
+            // Each guid has 16 bytes
+            RegistryKey allDeskSubKey = Registry.CurrentUser.OpenSubKey("SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\VirtualDesktops", false);
+            byte[] allDeskValue = (byte[])allDeskSubKey.GetValue("VirtualDesktopIDs");
+            availableDesktops.Clear();
+
+            if (allDeskValue != null)
+            {
+                int numberOfDesktops = allDeskValue.Length / 16;
+
+                for (int i = 0; i < numberOfDesktops; i++)
+                {
+                    byte[] guidArray = new byte[16];
+                    Array.ConstrainedCopy(allDeskValue, i * 16, guidArray, 0, 16);
+                    availableDesktops.Add(new Guid(guidArray));
+                }
+            }
+            else
+            {
+                Log.Error("Failed to read the list of existing desktops form registry.", typeof(VirtualDesktopHelper));
+            }
+
+            // Guid for current desktop
+            RegistryKey currentDeskSubKey = Registry.CurrentUser.OpenSubKey("SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\SessionInfo\\1\\VirtualDesktops", false);
```

Comment:
- Session id = process.getcurrent().sessionid

- if reg value doesn't exist the user hasen't switched. then desktop is desktop 1